react/components/SponsoredBadge.tsx

```diff
@@ -0,0 +1,45 @@
+import classNames from 'classnames'
+import React from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+import { IOMessage } from 'vtex.native-types'
+
+const CSS_HANDLES = ['sponsoredBadgeContainer', 'sponsoredBadgeText'] as const
+
+type Props = {
+  /**
+   * The message ID or string for the sponsored label.
+   */
+  label?: string
+
+  /**
+   * Whether to show the sponsored label text.
+   */
+  showLabel?: boolean
+}
+
+export const SponsoredBadge = ({
+  label = 'store/sponsoredBadge.title',
+  showLabel = false,
+}: Props) => {
+  const { handles } = useCssHandles(CSS_HANDLES)
+
+  if (!showLabel) return null
```

Comment:
since this boolean only checks if should show anything, we could remove it and check this before adding the SponsoredBadge to the tree, like this:
```
{showSponsoredBadge && (
  <SponsoredBadge label={sponsoredBadge?.label} />
)}
```

what do you think?

Reply:
I did it like this thinking that, in some cases, we actually want to render the div, but not the label itself (for positioning purposes). So it would be easier to change when that moment comes.

But, since this is not a need here, I'll change to your suggestion. Better to work with the present.

Reply:
Done https://github.com/vtex-apps/product-summary/pull/390/commits/6a79bd6927e350e0e00a6450e3ce28b4a26dcdea